src/api/flaskr/service/shifu/block_funcs.py

```
@@ -201,8 +211,22 @@ def save_block_list_internal(
+                    if (
+                        isinstance(block_dto.block_ui, OptionDto)
+                        and block_dto.block_ui.profile_id
+                    ):
+                        block_dto.profile_option_info = get_profile_option_info(
+                            app, block_dto.block_ui.profile_id, get_current_language()
+                        )
+                    elif (
+                        isinstance(block_dto.block_ui, TextInputDto)
+                        and block_dto.block_ui.profile_ids
+                    ):
+                        if len(block_dto.block_ui.profile_ids) == 1:
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)